drivers/pwm/pwm_nrfx.c

```diff
@@ -284,63 +290,71 @@ static int pwm_nrfx_init(const struct device *dev)
 		seq_values_ptr_get(dev)[i] = PWM_NRFX_CH_VALUE(0, inverted);
 	}
 
-	nrfx_err_t result = nrfx_pwm_init(&config->pwm,
-					  &config->initial_config,
-					  NULL,
-					  NULL);
-	if (result != NRFX_SUCCESS) {
-		LOG_ERR("Failed to initialize device: %s", dev->name);
-		return -EBUSY;
-	}
-
 	return 0;
 }
 
-#ifdef CONFIG_PM_DEVICE
-static void pwm_nrfx_uninit(const struct device *dev)
+static int pwm_suspend(const struct device *dev)
 {
 	const struct pwm_nrfx_config *config = dev->config;
+	struct pwm_nrfx_data *data = dev->data;
 
-	nrfx_pwm_uninit(&config->pwm);
+	if (data->active) {
+		/* PWM is still in use. Force stopping. */
+		data->stop_requested = true;
+		nrfx_pwm_stop(&config->pwm, false);
+		while (data->stop_requested) {
+		}
+	}
```

Comment:
This could be replaced with:
```
	if (!nrfx_pwm_stopped_check(&config->pwm)) {
		nrfx_pwm_stop(&config->pwm, true);
	}
```
to avoid introducing `active`.

Reply:
changed.